### DISCUSSION TITLE
Strapi doesn't allows keeping track of user's who created/updated any content

### DISCUSSION THREAD

**Discussion by vinod77** [strapi/strapi#6402] [Legacy Topics]
**Describe the issue**
Strapi doesn't allows keeping track of user's who created/updated any content

I want to know user who created & updated any record, but currently Strapi doesn't provides this feature or if i am wrong, please let me know how do i know about this.

Currently i can only see CreatedAt & UpdatedAt attributes by default which are working good.

**It will be great to have CreatedBy & UpdatedBy as well. This feature is really important for us.**

**Expected behavior**
Should be able to see & keep track of User who created & updated any record.

**System**

- Node.js version: v10.15.3
- NPM version: 6.4.1
- Strapi version: 3.0.0-beta.20.3
- Database: Mongo & S3 File storage for File uploads
- Operating system: MAC

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by derrickmehaffy** [comment]
You can easily create this on your own using the stable version (v3.0.0, not the beta).

Create a relation to users for both new fields (I would recommend a `oneWay` relation type, far left on the relation menu). Then you can create a lifecycle callback for `beforeCreate` and `beforeUpdate` following the documentation: https://strapi.io/documentation/v3.x/concepts/models.html#available-lifecycle-hooks to assign the user ID based on the incoming `ctx.stateu.user`.

Do however note you will not be able to assign an AdminUI "admin user" as those cannot have relations. This would only be for the API routes used by your frontend. For the admin users you will need to wait for the new RBAC feature coming around July of 2020. https://portal.productboard.com/strapi/1-public-roadmap/c/8-users-roles-permissions-ce-ee

I believe they intend to allow AdminUI "admin users" to be used as normal users via a provider to the users-permissions plugin. There is no work around until the new RBAC feature comes out due to the following issue: https://github.com/strapi/strapi/issues/3399

**Reply by jonathanmach** [reply]
Hi @derrickmehaffy , thanks for this message. How do I access the `ctx.stateu.user` from inside a `beforeCreate ` lifecycle?
All I can see available is the `data` received by the lifecycle.

**Comment by vinod77** [comment]
Hi @derrickmehaffy Thank you for your response.
I am able to do it by overriding `strapi-plugin-contentManager`

**Reply by vaibhavanand470** [reply]
can you give please some details on how you managed to achieve this. I have a similar use-case and i want to know which admin updated or created an entry

**Reply by vinod77** [reply]
Hi @vaibhavanand470 

1. I have updated model and added ```UpdatedBy: { type: string }```
2. In the admin panel i made this field non-editable
3. Override `content-manager/controllers/ContentManger.js` with below code.

```
ctx.body = await contentManagerService.create({
    ...data,  
    createdBy: `${ctx.state.user.firstName} ${ctx.state.user.lastName}`,
    updatedBy: `${ctx.state.user.firstName} ${ctx.state.user.lastName}`
  }, {
    files,
    model
});

```

CC: @derrickmehaffy

**Reply by vaibhavanand470** [reply]
Thanks @vinod77 for the reply. I managed to to do it in a similar way after reading your initial answer followed by some research. Kudos!
